R/memoise.r

```diff
@@ -95,8 +95,16 @@ memoise <- memoize <- function(f) {
       res
     }
   }
-  attr(memo_f, "memoised") <- TRUE
-  return(memo_f)
+  class(memo_f) <- c("memoised", "function")
+  memo_f
+}
+
+#' @export
+print.memoised <- function(x) {
+  cat("Memoised Function: ",
```

Comment:
Yes you are right, the current form was a holdover to my first iteration which used `message()` instead of `cat()`. I decided `cat()` was more consistent with the existing behavior, but did not re-simplify. Thanks!